### PR TITLE
Add datasources for machine and physical nic

### PR DESCRIPTION
**Pull request by ianmcmahon**
Create datasources for existing machines and physical nics

### CODE REVIEW

File: maas/provider.go [hunk 1]
```diff
@@ -53,6 +53,8 @@ func Provider() *schema.Provider {
 			"maas_fabric": dataSourceMaasFabric(),
 			"maas_vlan":   dataSourceMaasVlan(),
 			"maas_subnet": dataSourceMaasSubnet(),
+            "maas_machine": dataSourceMaasMachine(),
```
Comment (cpg1111) [inline maas/provider.go:56]: seems you need to run `go fmt`

**Review by cpg1111 — CHANGES_REQUESTED** [review]
One small issue with missing format, otherwise, looks good.

**Comment by skatsaounis** [comment]
Hi @ianmcmahon

I know it has been a while since you opened this PR but we are interested in adding these data sources to the provider. Would you still like to contribute to this PR or should we proceed with our implementation? If yes, feel free to reach out for help if needed :)

**Review by troyanov — APPROVED** [review]

File: maas/data_source_maas_machine.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+func dataSourceMaasMachine() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMachineRead,
+
+		Schema: map[string]*schema.Schema{
+			"power_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"power_parameters": {
```
Comment (skatsaounis) [inline maas/data_source_maas_machine.go:20]: This one will not work for machines with IPMI `power_parameters` that include `workaround_flags`. A similar approach to the `resource_maas_machine` should be followed here.

**Review by skatsaounis — CHANGES_REQUESTED** [review]

File: maas/data_source_maas_machine.go (hunk 2 above)
Comment (troyanov) [reply maas/data_source_maas_machine.go:20]: Good catch!

**Comment by ironashram** [comment]
@troyanov for my use case (and i suppose for many more) it would be needed to fetch the maas_machine datasource to with pxe_mac_address as input, since the hostname which is currently used here is not predictable in case the machine has been discovered automatically

My fork already has this, it may not be the cleanest but it works quite well

**Comment by skatsaounis** [comment]
@ironashram I incorporated most of what you have in your fork, since it makes a lot of sense. At last, we are ready to run our e2e tests and get some approval here :tada:

**Review by skatsaounis — APPROVED** [review]

**Review by cpg1111 — APPROVED** [review]
LGTM

**Pull request merged**